### PR TITLE
jit(arm64): support Add, Mul, Sub operations

### PR DESCRIPTION
**Pull request by mathetake**
part of #187

### CODE REVIEW

**Comment by mathetake** [comment]
ok will add the comments then ready!

File: wasm/jit/jit_arm64.go [hunk 1]
```diff
@@ -101,32 +108,96 @@ func (c *arm64Compiler) applyConstToRegisterInstruction(insturction obj.As, cons
...
-	// emits corresponding (at most) 4-instructions to load such large constants.
-	store.To.Offset = offset
-	store.From.Type = obj.TYPE_REG
-	store.From.Reg = source
-	c.addInstruction(store)
+// applyRegisterOffsetMemoryToRegisterInstruction adds an instruction where source operand is a memory location and destination is a register.
+// The differece from applyMemoryToRegisterInstruction is that here we specify the offset by a register rather than offset constant.
+func (c *arm64Compiler) applyRegisterOffsetMemoryToRegisterInstruction(insturction obj.As, baseRegister, offsetRegister, destinationRegister int16) (err error) {
+	inst := c.newProg()
+	inst.As = insturction
```
Comment (codefromthecrypt) [inline wasm/jit/jit_arm64.go:135]: s/insturction/instruction/g

File: wasm/jit/jit_arm64.go [hunk 2]
```diff
@@ -101,32 +108,96 @@ func (c *arm64Compiler) applyConstToRegisterInstruction(insturction obj.As, cons
...
-	store.To.Reg = baseRegister
-	// Note: in raw arm64 assembly, immediates larger than 16-bits
-	// are not supported, but the assembler takes care of this and
-	// emits corresponding (at most) 4-instructions to load such large constants.
-	store.To.Offset = offset
-	store.From.Type = obj.TYPE_REG
-	store.From.Reg = source
-	c.addInstruction(store)
+// applyRegisterOffsetMemoryToRegisterInstruction adds an instruction where source operand is a memory location and destination is a register.
+// The differece from applyMemoryToRegisterInstruction is that here we specify the offset by a register rather than offset constant.
```
Comment (codefromthecrypt) [inline wasm/jit/jit_arm64.go:132]: s/differece/difference/g

File: wasm/jit/jit_arm64_test.go [hunk 3]
```diff
@@ -24,20 +25,48 @@ func (j *jitEnv) requireNewCompiler(t *testing.T) *arm64Compiler {
 	return ret
 }
 
+// TODO: delete this as this could be a duplication from other tests especially spectests.
```
Comment (codefromthecrypt) [inline wasm/jit/jit_arm64_test.go:28]: good note

File: wasm/jit/jit_arm64_test.go [hunk 4]
```diff
@@ -184,3 +213,309 @@ func TestArm64Compiler_compileConsts(t *testing.T) {
...
+		{name: "int", stackPointer: 10, isFloat: false},
+		{name: "float", stackPointer: 10, isFloat: true},
+		{name: "int-huge-height", stackPointer: math.MaxInt16 + 1, isFloat: false},
+		{name: "float-huge-height", stackPointer: math.MaxInt16 + 1, isFloat: true},
+	} {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			env := newJITEnvironment()
+
+			// Build codes.
```
Comment (codefromthecrypt) [inline wasm/jit/jit_arm64_test.go:233]: s/codes/code/g

**Review by codefromthecrypt — APPROVED** [review]
LGTM

File: wasm/jit/jit_arm64_test.go [hunk 5]
```diff
@@ -184,3 +213,309 @@ func TestArm64Compiler_compileConsts(t *testing.T) {
...
+		wazeroir.OperationKindSub,
+		wazeroir.OperationKindMul,
+	} {
+		kind := kind
+		t.Run(kind.String(), func(t *testing.T) {
+			for _, unsignedType := range []wazeroir.UnsignedType{
+				wazeroir.UnsignedTypeI32,
+				wazeroir.UnsignedTypeI64,
+				wazeroir.UnsignedTypeF32,
+				wazeroir.UnsignedTypeF64,
```
Comment (nullpo-head) [inline wasm/jit/jit_arm64_test.go:366]: Just curious, why those floats are called `Unsigned` while they're actually signed?

**Review by nullpo-head — APPROVED** [review]
Looks great to me. Thanks for your work!

File: wasm/jit/jit_arm64_test.go (hunk 5 above)
Comment (mathetake) [reply wasm/jit/jit_arm64_test.go:366]: UnsignedType = UnsignedInteger || Floats = (I32, I64, F32, F64)!

Comment (mathetake) [reply wasm/jit/jit_arm64_test.go:366]: I think I will rework around this #210 !

**Comment by mathetake** [comment]
Thanks for the look!!

**Pull request merged**